imqsauth/http.go

```diff
@@ -776,6 +798,33 @@ func httpHandlerUpdateUser(central *ImqsCentral, w http.ResponseWriter, r *httpR
 	}
 }
 
+func httpHandlerUnlockUser(central *ImqsCentral, w http.ResponseWriter, r *httpRequest) {
+	username := strings.TrimSpace(r.http.URL.Query().Get("username"))
+
+	userId, err := getUserId(r)
```

Comment:
Wait a second!
Does this API call allow a user to unlock himself?
I see that it doesn't exactly, but the API has the handlerFlagNeedAdminRights flags on it, so that's OK. HOWEVER, this function is ignoring the "username" that is sent in via the URL, and instead it's unlocking the user that's making the call.

Reply:
I'm pretty sure that the userid in here is not the caller but a userId extracted from the request by getUserId

Reply:
Yep, you're right. I probably wrote that function, but I must say it could be named better. But all good, just leave it as-is.